### PR TITLE
Add https to apt sources on final image, actually apt upgrade...

### PR DESCRIPTION
**Pull request by gStart9**
and ensure 'torify' (torsocks) is installed

### CODE REVIEW

**Comment by gStart9** [comment]
Hi fellas.  I just saw this in my CLN logs:
```
2023-06-27T04:34:59+00:00  2023-06-27T04:34:59.981Z UNUSUAL plugin-clboss: DnsSeed: always-use-proxy set, but none of our known seeds could be accessed over `torify`. Is `torify` installed?
```

Checked and it seems though torsocks (aka torify) never made it to the final container, so I added that in (along with ca-certificates for good measure).  Another thing I did is have it make sure all the debian repos in /etc/apt/sources.list use https and then actually do an `apt -y upgrade` in every container along the way of building / compiling all these apps.  I'm not sure it's completely necessary but I figured if, for instance, when I see something like this:
```
root@c-lightning:/app# apt list --upgradable
Listing... Done
bind9-dnsutils/oldstable-security 1:9.16.42-1~deb11u1 amd64 [upgradable from: 1:9.16.37-1~deb11u1]
bind9-host/oldstable-security 1:9.16.42-1~deb11u1 amd64 [upgradable from: 1:9.16.37-1~deb11u1]
bind9-libs/oldstable-security 1:9.16.42-1~deb11u1 amd64 [upgradable from: 1:9.16.37-1~deb11u1]
dnsutils/oldstable-security 1:9.16.42-1~deb11u1 all [upgradable from: 1:9.16.37-1~deb11u1]
libssl1.1/oldstable-security 1.1.1n-0+deb11u5 amd64 [upgradable from: 1.1.1n-0+deb11u4]
openssl/oldstable-security 1.1.1n-0+deb11u5 amd64 [upgradable from: 1.1.1n-0+deb11u4]
```
That it's better if at each container in which software gets built has the newest security fixes like openssl, that would be good.  LMK if it's not a concern for some reason and I can remove them.  At the very least I'm pretty sure `apt -y upgrade` is a good step to take in the 'final' container.

**Comment by k0gen** [comment]
Good call to use SSL/TLS for fetching APT packages! However, incorporating `apt -y upgrade` within the Dockerfile is generally not considered a good practice. It adversely impacts the build time (which is already very long in this one) and, even worse, increases the image size. I won't mention the lack of determinism since we are already retrieving the latest debian-slim image every build and the updates you observe are just some minor ones. Notably, OpenSSL and Libssl will still remain at version 1.1.1n.

**Comment by gStart9** [comment]
I've removed the `apt upgrade`s.  Main point of this exercise was to make torify available for clboss, which it now should be.  I'll test today.

File: Dockerfile [hunk 1]
```diff
@@ -136,7 +135,7 @@ RUN git clone --recursive /tmp/lightning-wrapper/lightning . && \
     git checkout $(git --work-tree=/tmp/lightning-wrapper/lightning --git-dir=/tmp/lightning-wrapper/lightning/.git rev-parse HEAD)
     # git checkout $(git --git-dir=/tmp/lightning-wrapper/.git rev-parse HEAD)
 
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python3 - \
+RUN curl -sSL https://install.python-poetry.org | python3 - \
```
Comment (k0gen) [inline Dockerfile:138]: This is already in the master

**Review by Dominion5254 — APPROVED** [review]

**Comment by Dominion5254** [comment]
Closed pending further investigation of torify.

**Pull request closed without merging**